Update documentation (#19)
VirtusLab/graphbuddy: README.md

# Say hello to Graph Buddy!

## Your new best pal to help you understand your Java and Scala code better

With today's IDEs, we are all forced to browse code written as text in flatly-structured files, with
almost no information about the semantic dependencies between particular code units. What if we
could take a different look and, instead of seeing just source code in text files, go through

> colorful graph nodes that instantly and clearly show you dependencies and other essential structural

Author: yes, you are right

> colorful graph nodes that instantly and clearly show you dependencies and other essential structural

Author: How does it sound now?

> colorful graph nodes that instantly and clearly show you dependencies and other essential structural

Reviewer: LGTM

## Graph Buddy features

**NOTE: The base network mode for features is 2d, some features are not/less available on 3d
graph.**

### Adding nodes to graph

You can do this in multiple ways. The most intuitive is clicking on the code - then corresponding node should appear on plugin canvas.

Second way is using options from context menu in editor or on canvas. In editor there are some options to generate whole graph for selected place in code.

### Filters, layouts, search

You can filter elements of some type using menu on the top of window. Layouts can help you organize nodes on canvas in the way you like. And finally, using search, you can highlight elements on canvas which are matched to typed phrase like `get` or `class`.

### Saving state

If you have created graph, which you want to preserve on the future (f.e. to present it your colleague), you can do this by clicking on floppy disk icon on toolbar. There you can save snaphshot of your current work or load saved previously. You can find those snapshots in `.graphbuddy` folder.

### Regenerating graph files and reindexing

> It varies a bit depending on language do you use. If you use Java, then after f.e. changing branch some files may be outdated. Currently, this situation is not detected by plugin, so you should click `Generate graph` in such situation.

Reviewer: ```suggestion
If you are working with a Java project, then under some circumstances (switching branches, appling a code patch, editing or removing multiple files at once) some graph files may become outdated and as a result, the plugin may not work as expected (nodes and edges may be missing from the graph or pointing to incorrect locations in the code). There currently is no automated detection for these cases. When in doubt, you should trigger the `Generate graph` action from the Main Menu to make sure you are working with up-to-date graph files.
```

> If you have a Scala project, then after recompilation you should manually `reindex graph`. It will load newly generated files.

Reviewer: ```suggestion
If you are working with a Scala project, assuming you have configured the compiler plugin correctly, you should manually trigger the `reindex graph` action from the Main Menu. This will load the newly generated graph files to the plugin.
```

> ### Detailed options

Reviewer: ```suggestion
### Advanced options
```

<table>
  <tr>
    <td>Filters</td>
    <td>By clicking the buttons in filtering menu, you toggle visibility of corresponding node/edge kind</td>
  </tr>
  <tr>
    <td>Layouts</td>
    <td>By switching layout in the navigation bar (layers icon), you change rendering method of the graph (classic, vertical, horizontal)</td>
  </tr>
  <tr>
    <td>Search</td>
    <td>You can search for elements on the canvas, e.g. `Class`, `Method` or node name (like `get`).</td>
  </tr>
  <tr>
    <td>Settings</td>
    <td>By selecting additional network options, you can adjust graph visibility and amount of visible information to your liking. You can also switch the view between 2d and 3d.</td>
  </tr>
</table>

#### Interface buttons

<table>
  <tr>
    <td>Undo/Redo</td>

>     <td>Undo or redo action on graph</td>

Reviewer: ```suggestion
    <td>Undo or redo action on the graph</td>
```

  </tr>
  <tr>
    <td>Zoom</td>
    <td>Zoom in or out</td>
  </tr>
  <tr>
    <td>Reorganise graph</td>
    <td>By clicking reorganise graph button, you reorganise network nodes positions</td>
  </tr>
  <tr>
    <td>Freeze graph</td>

>     <td>By clicking the `freeze graph` button, you lock the graph, disallowing any updates to its data</td>

Reviewer: ```suggestion
    <td>By clicking the `freeze graph` button, you lock the graph, temporarily disallowing any updates to its data</td>
```

    <td>By pressing 'ctrl+z' you revert last action (add/update/remove/re-position)</td>
  </tr>
  <tr>
    <td>Redo action</td>

>     <td>By pressing 'ctrl+shift+z' you redo action (add/update/remove/re-position)</td>

Reviewer: ```suggestion
    <td>By pressing 'ctrl+shift+z' you redo the last undone action (add/update/remove/re-position)</td>
```

  </tr>
  <tr>
    <td>Close modal</td>

>     <td>By pressing 'esc' you close any modal currently active</td>

Reviewer: ```suggestion
    <td>By pressing 'esc' you close currently active modal</td>
```

  </tr>
</table>

### Navigating through the project

Each declaration or definition is represented as a _node_ and the connection between them as an

> _edge_. When browsing the code, Graph Buddy will update the graph board interactively. Additionally,

Reviewer: ```suggestion
_edge_. When browsing your code in the editor, Graph Buddy will update the graph canvas interactively. As your cursor is moved through the code, relevant pieces of the graph will be drawn on the canvas. Additionally,
```
